tests/components/squeezebox/test_config_flow.py

```diff
@@ -0,0 +1,263 @@
+
+
+async def test_discovery(hass):
+    """Test handling of discovered server."""
+    with patch(
+        "pysqueezebox.Server.async_query", return_value={"uuid": UUID},
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_DISCOVERY},
+            data={CONF_HOST: HOST, CONF_PORT: PORT, "uuid": UUID},
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
+
+
+async def test_discovery_no_uuid(hass):
+    """Test handling of discovered server with unavailable uuid."""
+    with patch("pysqueezebox.Server.async_query", new=patch_async_query_unauthorized):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_DISCOVERY},
+            data={CONF_HOST: HOST, CONF_PORT: PORT},
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
+
+
+async def test_import(hass):
+    """Test handling of configuration imported."""
+    with patch("pysqueezebox.Server.async_query", return_value={"uuid": UUID},), patch(
+        "homeassistant.components.squeezebox.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.squeezebox.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={CONF_HOST: HOST, CONF_PORT: PORT},
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+
+        await hass.async_block_till_done()
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_bad_host(hass):
+    """Test handling of configuration imported with bad host."""
+    with patch("pysqueezebox.Server.async_query", return_value=False):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={CONF_HOST: HOST, CONF_PORT: PORT},
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "cannot_connect"
+
+
+async def test_import_bad_auth(hass):
+    """Test handling of configuration import with bad authentication."""
+    with patch("pysqueezebox.Server.async_query", new=patch_async_query_unauthorized):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                CONF_HOST: HOST,
+                CONF_PORT: PORT,
+                CONF_USERNAME: "test",
+                CONF_PASSWORD: "bad",
+            },
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "invalid_auth"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)